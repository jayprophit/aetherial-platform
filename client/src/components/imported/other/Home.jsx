import React, { useState, useEffect } from 'react'
import { Link } from 'react-router-dom'
import { 
  ArrowRight, Play, Star, TrendingUp, Users, Globe, Zap, Shield,
  Brain, CreditCard, Heart, BookOpen, Smartphone, Wifi, Bluetooth,
  Radio, Satellite, Watch, Home as HomeIcon, Car, Lightbulb,
  Camera, Headphones, Gamepad2, Tv, Speaker, Thermometer,
  Lock, Eye, Mic, Video, MessageCircle, Phone, Mail,
  BarChart3, PieChart, LineChart, Activity, Clock, Award,
  Rocket, Target, Sparkles, Crown, Diamond, Gem
} from 'lucide-react'
import { Button } from '@/components/ui/button'

const Home = ({ user }) => {
  const [currentSlide, setCurrentSlide] = useState(0)
  const [liveStats, setLiveStats] = useState({
    users: 2547832,
    transactions: 125847293,
    countries: 195,
    uptime: 99.99
  })

  // Hero slides
  const heroSlides = [
    {
      title: "The Ultimate Unified Platform",
      subtitle: "AI, Blockchain, Banking, Healthcare, and 200+ Features in One Platform",
      image: "/api/placeholder/800/400",
      cta: "Get Started Free"
    },
    {
      title: "Smart Technology Ecosystem",
      subtitle: "5G/6G, IoT, Smart Wearables, Satellite Communication, and Future-Proof Tech",
      image: "/api/placeholder/800/400",
      cta: "Explore Technology"
    },
    {
      title: "Advanced Connectivity Hub",
      subtitle: "Voice, VPN, Bluetooth, Mesh Networks, and Quantum Communication",
      image: "/api/placeholder/800/400",
      cta: "Connect Everything"
    }
  ]

  // Platform features
  const platformFeatures = [
    {
      category: "AI & Intelligence",
      icon: Brain,
      color: "from-purple-500 to-pink-500",
      features: [
        "GPT-4, Claude-3, Gemini Pro Integration",
        "Interactive Chat with Code Execution",
        "Voice & Video AI Calls",
        "Multi-Modal Processing",
        "Quantum-Enhanced AI",
        "Real-Time Learning",
        "Custom Model Training",
        "AI-Powered Automation"
      ]
    },
    {
      category: "Smart Connectivity",
      icon: Wifi,
      color: "from-blue-500 to-cyan-500",
      features: [
        "5G/6G Network Optimization",
        "Satellite Communication",
        "Bluetooth Device Management",
        "Mesh Network Creation",
        "VPN & Secure Tunneling",
        "Voice Recognition & Commands",
        "Future-Proof Protocols",
        "Edge Computing Integration"
      ]
    },
    {
      category: "IoT & Smart Devices",
      icon: HomeIcon,
      color: "from-green-500 to-emerald-500",
      features: [
        "Smart Home Automation",
        "Wearable Device Integration",
        "Industrial IoT Control",
        "Smart City Infrastructure",
        "Automotive Connectivity",
        "Health Monitoring Devices",
        "Environmental Sensors",
        "Predictive Maintenance"
      ]
    },
    {
      category: "Blockchain & Finance",
      icon: Shield,
      color: "from-orange-500 to-red-500",
      features: [
        "3D Blockchain with Runestones",
        "Multi-Consensus Mechanisms",
        "DeFi Protocols & Flash Loans",
        "NFT Marketplace",
        "Quantum-Resistant Cryptography",
        "Cross-Chain Interoperability",
        "Smart Contract Automation",
        "Decentralized Identity"
      ]
    },
    {
      category: "Healthcare & Life Sciences",
      icon: Heart,
      color: "from-red-500 to-pink-500",
      features: [
        "AI Medical Diagnostics",
        "CRISPR Gene Editing Tools",
        "Telemedicine Platform",
        "Health Data Analytics",
        "Drug Discovery AI",
        "Clinical Trial Management",
        "Wearable Health Monitoring",
        "Personalized Medicine"
      ]
    },
    {
      category: "Business & Enterprise",
      icon: BarChart3,
      color: "from-indigo-500 to-purple-500",
      features: [
        "Complete ERP System",
        "CRM & Sales Automation",
        "Advanced Analytics",
        "Workflow Automation",
        "Supply Chain Management",
        "Financial Planning",
        "Team Collaboration",
        "Business Intelligence"
      ]
    }
  ]

  // Smart devices showcase
  const smartDevices = [
    { name: "Smart Watch", icon: Watch, status: "Connected", battery: 85 },
    { name: "Smart Phone", icon: Smartphone, status: "Connected", battery: 92 },
    { name: "Smart Home", icon: HomeIcon, status: "Active", devices: 24 },
    { name: "Smart Car", icon: Car, status: "Parked", range: "285 miles" },
    { name: "Smart Lights", icon: Lightbulb, status: "Auto", brightness: 75 },
    { name: "Security Camera", icon: Camera, status: "Recording", quality: "4K" },
    { name: "Smart Speaker", icon: Speaker, status: "Listening", volume: 60 },
    { name: "Thermostat", icon: Thermometer, status: "Auto", temp: "72Â°F" }
  ]

  // Connectivity features
  const connectivityFeatures = [
    { name: "5G Network", icon: Radio, status: "Ultra Fast", speed: "2.1 Gbps" },
    { name: "6G Ready", icon: Zap, status: "Future-Proof", speed: "100+ Gbps" },
    { name: "Satellite", icon: Satellite, status: "Global", coverage: "100%" },
    { name: "Bluetooth", icon: Bluetooth, status: "Connected", devices: 8 },
    { name: "WiFi 7", icon: Wifi, status: "Active", speed: "46 Gbps" },
    { name: "VPN Shield", icon: Lock, status: "Protected", servers: 150 }
  ]

  // Live stats animation
  useEffect(() => {
    const interval = setInterval(() => {
      setLiveStats(prev => ({
        users: prev.users + Math.floor(Math.random() * 50),
        transactions: prev.transactions + Math.floor(Math.random() * 500),
        countries: 195,
        uptime: 99.99
      }))
    }, 3000)
    return () => clearInterval(interval)
  }, [])

  // Auto-slide hero
  useEffect(() => {
    const interval = setInterval(() => {
      setCurrentSlide((prev) => (prev + 1) % heroSlides.length)
    }, 5000)
    return () => clearInterval(interval)
  }, [])

  return (
    <div className="min-h-screen bg-gradient-to-br from-slate-50 via-blue-50 to-purple-50 dark:from-gray-900 dark:via-blue-900 dark:to-purple-900">
      {/* Hero Section */}
      <section className="relative h-screen flex items-center justify-center overflow-hidden">
        {/* Background Animation */}
        <div className="absolute inset-0 bg-gradient-to-r from-blue-600/20 to-purple-600/20">
          <div className="absolute inset-0 opacity-20 animate-pulse">
            <div className="w-full h-full bg-gradient-to-br from-blue-400 to-purple-400 rounded-full blur-3xl" />
          </div>
        </div>

        <div className="relative z-10 max-w-7xl mx-auto px-4 text-center">
          {/* Hero Content */}
          <div className="mb-8">
            <h1 className="text-6xl md:text-8xl font-bold mb-6">
              <span className="bg-gradient-to-r from-blue-600 via-purple-600 to-pink-600 bg-clip-text text-transparent">
                Unified Platform
              </span>
            </h1>
            <p className="text-xl md:text-2xl text-gray-600 dark:text-gray-300 mb-8 max-w-4xl mx-auto">
              {heroSlides[currentSlide].subtitle}
            </p>
          </div>

          {/* Live Stats */}
          <div className="grid grid-cols-2 md:grid-cols-4 gap-6 mb-12">
            <div className="bg-white/20 dark:bg-gray-800/20 backdrop-blur-md rounded-2xl p-6 border border-white/30">
              <div className="text-3xl font-bold text-blue-600">{liveStats.users.toLocaleString()}+</div>
              <div className="text-sm text-gray-600 dark:text-gray-300">Active Users</div>
            </div>
            <div className="bg-white/20 dark:bg-gray-800/20 backdrop-blur-md rounded-2xl p-6 border border-white/30">
              <div className="text-3xl font-bold text-green-600">{liveStats.transactions.toLocaleString()}+</div>
              <div className="text-sm text-gray-600 dark:text-gray-300">Transactions</div>
            </div>
            <div className="bg-white/20 dark:bg-gray-800/20 backdrop-blur-md rounded-2xl p-6 border border-white/30">
              <div className="text-3xl font-bold text-purple-600">{liveStats.countries}+</div>
              <div className="text-sm text-gray-600 dark:text-gray-300">Countries</div>
            </div>
            <div className="bg-white/20 dark:bg-gray-800/20 backdrop-blur-md rounded-2xl p-6 border border-white/30">
              <div className="text-3xl font-bold text-orange-600">{liveStats.uptime}%</div>
              <div className="text-sm text-gray-600 dark:text-gray-300">Uptime</div>
            </div>
          </div>

          {/* CTA Buttons */}
          <div className="flex flex-col sm:flex-row gap-4 justify-center items-center mb-12">
            {user ? (
              <Link to="/ai">
                <Button size="lg" className="px-8 py-4 text-lg">
                  <Rocket className="w-5 h-5 mr-2" />
                  Launch Platform
                </Button>
              </Link>
            ) : (
              <Button size="lg" className="px-8 py-4 text-lg">
                <Sparkles className="w-5 h-5 mr-2" />
                Get Started Free
              </Button>
            )}
            <Button variant="outline" size="lg" className="px-8 py-4 text-lg">
              <Play className="w-5 h-5 mr-2" />
              Watch Demo
            </Button>
          </div>

          {/* Hero Slide Indicators */}
          <div className="flex justify-center space-x-2">
            {heroSlides.map((_, index) => (
              <button
                key={index}
                onClick={() => setCurrentSlide(index)}
                className={`w-3 h-3 rounded-full transition-colors ${
                  index === currentSlide ? 'bg-blue-600' : 'bg-gray-300'
                }`}
              />
            ))}
          </div>
        </div>
      </section>

      {/* Smart Connectivity Section */}
      <section className="py-20 bg-white dark:bg-gray-800">
        <div className="max-w-7xl mx-auto px-4">
          <div className="text-center mb-16">
            <h2 className="text-4xl font-bold mb-4">
              <span className="bg-gradient-to-r from-blue-600 to-purple-600 bg-clip-text text-transparent">
                Smart Connectivity Ecosystem
              </span>
            </h2>
            <p className="text-xl text-gray-600 dark:text-gray-300 max-w-3xl mx-auto">
              Future-proof connectivity with 5G/6G, satellite, IoT, and quantum communication
            </p>
          </div>

          <div className="grid grid-cols-2 md:grid-cols-3 lg:grid-cols-6 gap-6 mb-16">
            {connectivityFeatures.map((feature, index) => (
              <div key={index} className="bg-gradient-to-br from-blue-50 to-purple-50 dark:from-gray-700 dark:to-gray-600 rounded-2xl p-6 text-center hover:scale-105 transition-transform">
                <feature.icon className="w-8 h-8 mx-auto mb-3 text-blue-600" />
                <div className="font-semibold text-sm mb-1">{feature.name}</div>
                <div className="text-xs text-green-600 mb-1">{feature.status}</div>
                <div className="text-xs text-gray-500">{feature.speed || feature.coverage || feature.devices || feature.servers}</div>
              </div>
            ))}
          </div>

          {/* Smart Devices Grid */}
          <div className="grid grid-cols-2 md:grid-cols-4 lg:grid-cols-8 gap-4">
            {smartDevices.map((device, index) => (
              <div key={index} className="bg-gray-50 dark:bg-gray-700 rounded-xl p-4 text-center hover:bg-gray-100 dark:hover:bg-gray-600 transition-colors">
                <device.icon className="w-6 h-6 mx-auto mb-2 text-gray-600 dark:text-gray-300" />
                <div className="text-xs font-medium mb-1">{device.name}</div>
                <div className="text-xs text-green-600">{device.status}</div>
                {device.battery && (
                  <div className="text-xs text-gray-500">{device.battery}%</div>
                )}
                {device.devices && (
                  <div className="text-xs text-gray-500">{device.devices} devices</div>
                )}
              </div>
            ))}
          </div>
        </div>
      </section>

      {/* Platform Features */}
      <section className="py-20 bg-gray-50 dark:bg-gray-900">
        <div className="max-w-7xl mx-auto px-4">
          <div className="text-center mb-16">
            <h2 className="text-4xl font-bold mb-4">
              <span className="bg-gradient-to-r from-purple-600 to-pink-600 bg-clip-text text-transparent">
                Comprehensive Platform Features
              </span>
            </h2>
            <p className="text-xl text-gray-600 dark:text-gray-300 max-w-3xl mx-auto">
              Everything you need in one unified platform - from AI to blockchain, healthcare to IoT
            </p>
          </div>

          <div className="grid grid-cols-1 md:grid-cols-2 lg:grid-cols-3 gap-8">
            {platformFeatures.map((category, index) => (
              <div key={index} className="bg-white dark:bg-gray-800 rounded-2xl p-8 shadow-lg hover:shadow-xl transition-shadow">
                <div className={`w-16 h-16 rounded-2xl bg-gradient-to-r ${category.color} flex items-center justify-center mb-6`}>
                  <category.icon className="w-8 h-8 text-white" />
                </div>
                <h3 className="text-2xl font-bold mb-4">{category.category}</h3>
                <ul className="space-y-2">
                  {category.features.map((feature, featureIndex) => (
                    <li key={featureIndex} className="flex items-center text-gray-600 dark:text-gray-300">
                      <div className="w-2 h-2 bg-blue-500 rounded-full mr-3" />
                      {feature}
                    </li>
                  ))}
                </ul>
                <Button className="w-full mt-6" variant="outline">
                  Explore {category.category}
                  <ArrowRight className="w-4 h-4 ml-2" />
                </Button>
              </div>
            ))}
          </div>
        </div>
      </section>

      {/* Voice & Communication Features */}
      <section className="py-20 bg-white dark:bg-gray-800">
        <div className="max-w-7xl mx-auto px-4">
          <div className="text-center mb-16">
            <h2 className="text-4xl font-bold mb-4">
              <span className="bg-gradient-to-r from-green-600 to-blue-600 bg-clip-text text-transparent">
                Advanced Communication Hub
              </span>
            </h2>
            <p className="text-xl text-gray-600 dark:text-gray-300 max-w-3xl mx-auto">
              Voice commands, video calls, messaging, email, and quantum-secure communication
            </p>
          </div>

          <div className="grid grid-cols-2 md:grid-cols-4 gap-6">
            <div className="bg-gradient-to-br from-blue-50 to-purple-50 dark:from-gray-700 dark:to-gray-600 rounded-2xl p-6 text-center">
              <Mic className="w-12 h-12 mx-auto mb-4 text-blue-600" />
              <h3 className="font-bold mb-2">Voice Control</h3>
              <p className="text-sm text-gray-600 dark:text-gray-300">Natural language commands</p>
            </div>
            <div className="bg-gradient-to-br from-green-50 to-blue-50 dark:from-gray-700 dark:to-gray-600 rounded-2xl p-6 text-center">
              <Video className="w-12 h-12 mx-auto mb-4 text-green-600" />
              <h3 className="font-bold mb-2">Video Calls</h3>
              <p className="text-sm text-gray-600 dark:text-gray-300">HD video conferencing</p>
            </div>
            <div className="bg-gradient-to-br from-purple-50 to-pink-50 dark:from-gray-700 dark:to-gray-600 rounded-2xl p-6 text-center">
              <MessageCircle className="w-12 h-12 mx-auto mb-4 text-purple-600" />
              <h3 className="font-bold mb-2">Messaging</h3>
              <p className="text-sm text-gray-600 dark:text-gray-300">Secure instant messaging</p>
            </div>
            <div className="bg-gradient-to-br from-orange-50 to-red-50 dark:from-gray-700 dark:to-gray-600 rounded-2xl p-6 text-center">
              <Mail className="w-12 h-12 mx-auto mb-4 text-orange-600" />
              <h3 className="font-bold mb-2">Email</h3>
              <p className="text-sm text-gray-600 dark:text-gray-300">Integrated email system</p>
            </div>
          </div>
        </div>
      </section>

      {/* CTA Section */}
      <section className="py-20 bg-gradient-to-r from-blue-600 to-purple-600">
        <div className="max-w-4xl mx-auto px-4 text-center text-white">
          <h2 className="text-4xl font-bold mb-6">Ready to Experience the Future?</h2>
          <p className="text-xl mb-8 opacity-90">
            Join millions of users already using the most comprehensive platform ever created
          </p>
          <div className="flex flex-col sm:flex-row gap-4 justify-center">
            <Button size="lg" variant="secondary" className="px-8 py-4 text-lg">
              <Crown className="w-5 h-5 mr-2" />
              Start Free Trial
            </Button>
            <Button size="lg" variant="outline" className="px-8 py-4 text-lg border-white text-white hover:bg-white hover:text-blue-600">
              <Diamond className="w-5 h-5 mr-2" />
              View Pricing
            </Button>
          </div>
        </div>
      </section>
    </div>
  )
}

export default Home

