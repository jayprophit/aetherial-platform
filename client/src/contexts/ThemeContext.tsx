import React, { createContext, useContext, useEffect, useState } from "react";

type Theme = "light" | "dark";

interface ThemeContextType {
  theme: Theme;
  toggleTheme?: () => void;
  switchable: boolean;
}

const ThemeContext = createContext<ThemeContextType | undefined>(undefined);

interface ThemeProviderProps {
  children: React.ReactNode;
  defaultTheme?: Theme;
  switchable?: boolean;
}

export function ThemeProvider({
  children,
  defaultTheme = "light",
  switchable = true, // Enable by default
}: ThemeProviderProps) {
  const [theme, setTheme] = useState<Theme>(() => {
    if (switchable) {
      // Check localStorage first
      const stored = localStorage.getItem("theme");
      if (stored) return stored as Theme;
      
      // Then check system preference
      if (window.matchMedia && window.matchMedia('(prefers-color-scheme: dark)').matches) {
        return "dark";
      }
    }
    return defaultTheme;
  });

  useEffect(() => {
    const root = document.documentElement;
    
    // Set data-theme attribute for CSS variables
    root.setAttribute('data-theme', theme);
    
    // Also set class for Tailwind dark mode
    if (theme === "dark") {
      root.classList.add("dark");
    } else {
      root.classList.remove("dark");
    }

    if (switchable) {
      localStorage.setItem("theme", theme);
    }
  }, [theme, switchable]);
  
  // Listen for system theme changes
  useEffect(() => {
    if (!switchable) return;
    
    const mediaQuery = window.matchMedia('(prefers-color-scheme: dark)');
    const handleChange = (e: MediaQueryListEvent) => {
      // Only auto-switch if user hasn't manually set a preference
      if (!localStorage.getItem("theme")) {
        setTheme(e.matches ? "dark" : "light");
      }
    };
    
    mediaQuery.addEventListener('change', handleChange);
    return () => mediaQuery.removeEventListener('change', handleChange);
  }, [switchable]);

  const toggleTheme = switchable
    ? () => {
        setTheme(prev => (prev === "light" ? "dark" : "light"));
      }
    : undefined;

  return (
    <ThemeContext.Provider value={{ theme, toggleTheme, switchable }}>
      {children}
    </ThemeContext.Provider>
  );
}

export function useTheme() {
  const context = useContext(ThemeContext);
  if (!context) {
    throw new Error("useTheme must be used within ThemeProvider");
  }
  return context;
}
