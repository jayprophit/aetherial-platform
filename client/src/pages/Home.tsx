import { useState } from 'react';
import { Heart, MessageCircle, Share2, MoreHorizontal, Image as ImageIcon, Video, Smile } from 'lucide-react';
import { Button } from '@/components/ui/button';
import { Card, CardContent, CardHeader } from '@/components/ui/card';
import { Textarea } from '@/components/ui/textarea';
import { Avatar, AvatarFallback, AvatarImage } from '@/components/ui/avatar';
import { useAuth } from '@/_core/hooks/useAuth';

// Mock data for posts
const mockPosts = [
  {
    id: 1,
    author: {
      name: 'Sarah Johnson',
      avatar: 'https://api.dicebear.com/7.x/avataaars/svg?seed=Sarah',
      username: '@sarahj'
    },
    content: 'Just completed my first course on blockchain development! 🎓 The Aetherial learning platform is amazing. Earned 50 AETH tokens as a reward!',
    image: 'https://images.unsplash.com/photo-1639762681485-074b7f938ba0?w=600&h=400&fit=crop',
    timestamp: '2 hours ago',
    likes: 24,
    comments: 5,
    shares: 2
  },
  {
    id: 2,
    author: {
      name: 'Michael Chen',
      avatar: 'https://api.dicebear.com/7.x/avataaars/svg?seed=Michael',
      username: '@mchen'
    },
    content: 'My new AI agent just made its first sale! 🤖💰 Automated e-commerce is the future. Check out the AI Agents marketplace if you want to create your own!',
    timestamp: '5 hours ago',
    likes: 42,
    comments: 12,
    shares: 8
  },
  {
    id: 3,
    author: {
      name: 'Emma Davis',
      avatar: 'https://api.dicebear.com/7.x/avataaars/svg?seed=Emma',
      username: '@emmad'
    },
    content: 'Just set up my smart home with Aetherial IoT! All my devices are now connected and earning me passive AETH. The automation features are incredible! 🏠✨',
    image: 'https://images.unsplash.com/photo-1558002038-1055907df827?w=600&h=400&fit=crop',
    timestamp: '1 day ago',
    likes: 67,
    comments: 18,
    shares: 15
  }
];

export default function Home() {
  const { user, isAuthenticated } = useAuth();
  const [postContent, setPostContent] = useState('');

  const handlePost = () => {
    // TODO: Implement post creation
    console.log('Creating post:', postContent);
    setPostContent('');
  };

  return (
    <div className="max-w-2xl mx-auto space-y-6">
      {/* Post Composer */}
      {isAuthenticated && (
        <Card>
          <CardHeader>
            <div className="flex items-start gap-4">
              <Avatar>
                <AvatarImage src={`https://api.dicebear.com/7.x/avataaars/svg?seed=${user?.name}`} />
                <AvatarFallback>{user?.name?.[0] || 'U'}</AvatarFallback>
              </Avatar>
              <div className="flex-1">
                <Textarea
                  placeholder="What's on your mind?"
                  value={postContent}
                  onChange={(e) => setPostContent(e.target.value)}
                  className="min-h-[100px] resize-none border-0 focus-visible:ring-0 p-0"
                />
              </div>
            </div>
          </CardHeader>
          <CardContent>
            <div className="flex items-center justify-between">
              <div className="flex gap-2">
                <Button variant="ghost" size="sm" className="gap-2">
                  <ImageIcon className="h-4 w-4" />
                  Photo
                </Button>
                <Button variant="ghost" size="sm" className="gap-2">
                  <Video className="h-4 w-4" />
                  Video
                </Button>
                <Button variant="ghost" size="sm" className="gap-2">
                  <Smile className="h-4 w-4" />
                  Emoji
                </Button>
              </div>
              <Button onClick={handlePost} disabled={!postContent.trim()}>
                Post
              </Button>
            </div>
          </CardContent>
        </Card>
      )}

      {/* Activity Feed */}
      <div className="space-y-4">
        {mockPosts.map((post) => (
          <Card key={post.id}>
            <CardHeader>
              <div className="flex items-start justify-between">
                <div className="flex items-center gap-3">
                  <Avatar>
                    <AvatarImage src={post.author.avatar} />
                    <AvatarFallback>{post.author.name[0]}</AvatarFallback>
                  </Avatar>
                  <div>
                    <h3 className="font-semibold">{post.author.name}</h3>
                    <p className="text-sm text-gray-500">{post.timestamp}</p>
                  </div>
                </div>
                <Button variant="ghost" size="icon">
                  <MoreHorizontal className="h-5 w-5" />
                </Button>
              </div>
            </CardHeader>
            <CardContent className="space-y-4">
              <p className="text-gray-800">{post.content}</p>
              
              {post.image && (
                <img
                  src={post.image}
                  alt="Post content"
                  className="w-full rounded-lg object-cover max-h-96"
                />
              )}

              <div className="flex items-center justify-between pt-4 border-t">
                <Button variant="ghost" size="sm" className="gap-2">
                  <Heart className="h-4 w-4" />
                  <span>{post.likes}</span>
                </Button>
                <Button variant="ghost" size="sm" className="gap-2">
                  <MessageCircle className="h-4 w-4" />
                  <span>{post.comments}</span>
                </Button>
                <Button variant="ghost" size="sm" className="gap-2">
                  <Share2 className="h-4 w-4" />
                  <span>{post.shares}</span>
                </Button>
              </div>
            </CardContent>
          </Card>
        ))}
      </div>
    </div>
  );
}

