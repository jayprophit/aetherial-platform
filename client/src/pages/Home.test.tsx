import { render } from '@testing-library/react';
import { AuthProvider } from '../contexts/AuthContext';
import { I18nextProvider } from 'react-i18next';
import i18n from '../i18n';
import Home from './Home';

test('renders the home page', () => {
  const { asFragment } = render(
    <I18nextProvider i18n={i18n}>
      <AuthProvider>
        <Home />
      </AuthProvider>
    </I18nextProvider>
  );
  expect(asFragment()).toMatchSnapshot();
});

