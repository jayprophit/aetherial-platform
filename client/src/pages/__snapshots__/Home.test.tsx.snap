// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`renders the home page 1`] = `
<DocumentFragment>
  <div
    class="max-w-4xl mx-auto px-4 py-6"
  >
    <div
      class="flex justify-end mb-4"
    >
      <button
        class="mr-2 px-3 py-1 rounded-md bg-gray-200 text-sm font-medium"
      >
        English
      </button>
      <button
        class="px-3 py-1 rounded-md bg-gray-200 text-sm font-medium"
      >
        Espa√±ol
      </button>
    </div>
    <div
      class="bg-white rounded-xl shadow-sm border border-slate-200 p-4 mb-6"
    >
      <div
        class="flex gap-3"
      >
        <img
          alt="Your avatar"
          class="w-10 h-10 rounded-full"
          src="https://api.dicebear.com/7.x/avataaars/svg?seed=guest"
        />
        <div
          class="flex-1"
        >
          <textarea
            class="w-full px-4 py-3 bg-slate-50 rounded-lg border border-slate-200 focus:outline-none focus:ring-2 focus:ring-cyan-500 focus:border-transparent resize-none"
            disabled=""
            placeholder="please_login_to_post"
            rows="3"
          />
          <p
            class="text-red-500 text-sm mt-2"
          >
            Failed to load posts
          </p>
          <div
            class="flex items-center justify-between mt-3"
          >
            <div
              class="flex gap-2"
            >
              <button
                class="p-2 hover:bg-slate-100 rounded-lg transition-colors"
                title="Add image"
              >
                <svg
                  class="lucide lucide-image w-5 h-5 text-slate-600"
                  fill="none"
                  height="24"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <rect
                    height="18"
                    rx="2"
                    ry="2"
                    width="18"
                    x="3"
                    y="3"
                  />
                  <circle
                    cx="9"
                    cy="9"
                    r="2"
                  />
                  <path
                    d="m21 15-3.086-3.086a2 2 0 0 0-2.828 0L6 21"
                  />
                </svg>
              </button>
              <button
                class="p-2 hover:bg-slate-100 rounded-lg transition-colors"
                title="Add video"
              >
                <svg
                  class="lucide lucide-video w-5 h-5 text-slate-600"
                  fill="none"
                  height="24"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="m16 13 5.223 3.482a.5.5 0 0 0 .777-.416V7.87a.5.5 0 0 0-.752-.432L16 10.5"
                  />
                  <rect
                    height="12"
                    rx="2"
                    width="14"
                    x="2"
                    y="6"
                  />
                </svg>
              </button>
              <button
                class="p-2 hover:bg-slate-100 rounded-lg transition-colors"
                title="Add emoji"
              >
                <svg
                  class="lucide lucide-smile w-5 h-5 text-slate-600"
                  fill="none"
                  height="24"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <circle
                    cx="12"
                    cy="12"
                    r="10"
                  />
                  <path
                    d="M8 14s1.5 2 4 2 4-2 4-2"
                  />
                  <line
                    x1="9"
                    x2="9.01"
                    y1="9"
                    y2="9"
                  />
                  <line
                    x1="15"
                    x2="15.01"
                    y1="9"
                    y2="9"
                  />
                </svg>
              </button>
              <button
                class="p-2 hover:bg-slate-100 rounded-lg transition-colors"
                title="Add location"
              >
                <svg
                  class="lucide lucide-map-pin w-5 h-5 text-slate-600"
                  fill="none"
                  height="24"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M20 10c0 4.993-5.539 10.193-7.399 11.799a1 1 0 0 1-1.202 0C9.539 20.193 4 14.993 4 10a8 8 0 0 1 16 0"
                  />
                  <circle
                    cx="12"
                    cy="10"
                    r="3"
                  />
                </svg>
              </button>
            </div>
            <button
              class="px-6 py-2 bg-gradient-to-r from-cyan-500 to-purple-600 text-white rounded-lg font-semibold hover:from-cyan-600 hover:to-purple-700 transition-all disabled:opacity-50 disabled:cursor-not-allowed flex items-center gap-2"
              disabled=""
            >
              Post
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      class="grid grid-cols-1 md:grid-cols-3 gap-4 mb-6"
    >
      <div
        class="bg-gradient-to-br from-cyan-500 to-cyan-600 rounded-xl p-4 text-white"
      >
        <div
          class="flex items-center justify-between"
        >
          <div>
            <p
              class="text-cyan-100 text-sm"
            >
              AETH Balance
            </p>
            <p
              class="text-2xl font-bold"
            >
              2,450
            </p>
          </div>
          <svg
            class="lucide lucide-zap w-8 h-8 text-cyan-200"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M4 14a1 1 0 0 1-.78-1.63l9.9-10.2a.5.5 0 0 1 .86.46l-1.92 6.02A1 1 0 0 0 13 10h7a1 1 0 0 1 .78 1.63l-9.9 10.2a.5.5 0 0 1-.86-.46l1.92-6.02A1 1 0 0 0 11 14z"
            />
          </svg>
        </div>
      </div>
      <div
        class="bg-gradient-to-br from-purple-500 to-purple-600 rounded-xl p-4 text-white"
      >
        <div
          class="flex items-center justify-between"
        >
          <div>
            <p
              class="text-purple-100 text-sm"
            >
              Friends
            </p>
            <p
              class="text-2xl font-bold"
            >
              342
            </p>
          </div>
          <svg
            class="lucide lucide-users w-8 h-8 text-purple-200"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M16 21v-2a4 4 0 0 0-4-4H6a4 4 0 0 0-4 4v2"
            />
            <circle
              cx="9"
              cy="7"
              r="4"
            />
            <path
              d="M22 21v-2a4 4 0 0 0-3-3.87"
            />
            <path
              d="M16 3.13a4 4 0 0 1 0 7.75"
            />
          </svg>
        </div>
      </div>
      <div
        class="bg-gradient-to-br from-pink-500 to-pink-600 rounded-xl p-4 text-white"
      >
        <div
          class="flex items-center justify-between"
        >
          <div>
            <p
              class="text-pink-100 text-sm"
            >
              Rank
            </p>
            <p
              class="text-2xl font-bold"
            >
              #127
            </p>
          </div>
          <svg
            class="lucide lucide-award w-8 h-8 text-pink-200"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="m15.477 12.89 1.515 8.526a.5.5 0 0 1-.81.47l-3.58-2.687a1 1 0 0 0-1.197 0l-3.586 2.686a.5.5 0 0 1-.81-.469l1.514-8.526"
            />
            <circle
              cx="12"
              cy="8"
              r="6"
            />
          </svg>
        </div>
      </div>
    </div>
    <div
      class="bg-white rounded-xl shadow-sm border border-slate-200 p-12 text-center"
    >
      <svg
        class="lucide lucide-trending-up w-16 h-16 text-slate-300 mx-auto mb-4"
        fill="none"
        height="24"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <polyline
          points="22 7 13.5 15.5 8.5 10.5 2 17"
        />
        <polyline
          points="16 7 22 7 22 13"
        />
      </svg>
      <h3
        class="text-xl font-semibold text-slate-800 mb-2"
      >
        no_posts_yet
      </h3>
      <p
        class="text-slate-600"
      >
        be_the_first_to_share
      </p>
    </div>
  </div>
</DocumentFragment>
`;
